pkgs/by-name/fa/fawltydeps/package.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  writableTmpDirAsHomeHook,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "fawltydeps";
+  version = "0.20.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "tweag";
+    repo = "FawltyDeps";
+    tag = "v${version}";
+    hash = "sha256-RGwCi4SD0khuOZXcR9Leh9WtRautnlJIfuLBnosyUgk=";
+  };
+
+  build-system = with python3Packages; [ poetry-core ];
+
+  dependencies = with python3Packages; [
+    pyyaml
+    importlib-metadata
+    isort
+    pip-requirements-parser
+    pydantic
+  ];
+
+  nativeCheckInputs =
+    [
+      writableTmpDirAsHomeHook
+    ]
+    ++ (with python3Packages; [
+      pytestCheckHook
+      hypothesis
+    ]);
+
+  disabledTestPaths = [
+    # Try to access files.pythonhosted.org
+    "tests/test_install_deps.py"
+    "tests/test_local_env.py"
+    "tests/test_packages.py"
+    "tests/test_resolver.py"
+    "tests/test_settings.py"
+    "tests/test_traverse_project.py"
+    "tests/test_types.py"
+  ];
```

Comment:
I'll have a look if anything can be done upstream to make these tests more nix-friendly...